docs/src/tutorials/yocto-linux.md

```diff
@@ -51,61 +50,191 @@ process. If using these options, to see previous commits or other branches see
+    ├── meta-skeleton
+    ├── meta-yocto-bsp
+    ├── oe-init-build-env
+    ├── README.hardware.md -> meta-yocto-bsp/README.hardware.md
+    ├── README.md -> README.poky.md
+    ├── README.OE-Core.md
+    ├── README.poky.md -> meta-poky/README.poky.md
+    ├── README.qemu.md
+    └── scripts
+```
+
+### Initialize the build environment
+
+To use the `bitbake`, `bitbake-layers`, `runqemu`, or other tools used in yocto to create or run our build, we need to
+source the `poky/oe-init-build-env` script in our shell:
+
 ```
-$ git clone git://git.yoctoproject.org/poky --branch=kirkstone --depth=1
+cd poky
+source oe-init-build-env
 ```
 
+The script creates a `build` directory, which itself contains only a `conf` directory:
+
+```
+.
+└── poky
+    ├── build
+    │   └── conf
+    │       ├── bblayers.conf
+    │       ├── local.conf
+    │       └── templateconf.cfg
+    ...
+```
+
+Inside the `poky/build/conf` directory, there are 2 files of interest to us:
+
+- `bblayers.conf` contains all the layers used by build
+- `local.conf` contains local build configuration, ie. configuration that applies only to the build
+
+In the following steps, we will edit these files to customize our image.
+
 ### Add layers
 
-We'll be using `meta-tedge` and `meta-openembedded`. First, fetch the repositories:
+Clone the `meta-tedge` and `meta-openembedded` repositories to our working directory:
 
 ```bash
 git clone https://github.com/thin-edge/meta-tedge
-git clone git://git.openembedded.org/meta-openembedded
+git clone --branch=kirkstone git://git.openembedded.org/meta-openembedded
 ```
 
-Next, add the following layers:
+> As with Poky itself, when cloning meta-openembedded either provide `--branch=kirkstone` for the clone command or
+> manually `git switch kirkstone` in `meta-openembedded` repository after cloning. Some Yocto layers support different
+> Yocto versions on different branches, in which case be sure to select correct branch. `meta-tedge` however supports 2
+> versions, Honister and Kirkstone on the main branch, so there's no need to change the default branch.
+
+Resulting in the following directory structure:
+```
+.
+├── meta-openembedded
+├── meta-tedge
+└── poky
+```
+
+> As these layers and `poky` are all different git repositories, we clone them next to the poky directory, but you can put
+> them inside `poky` if you prefer. The only thing that matters is to use a correct path in the
+> `poky/build/conf/bblayers.conf` file.
+
+Next, we add these layers to our build using `bitbake-layers` tool. Be aware that `meta-openembedded` is not itself a
+layer, but a collection of many layers:
 
 ```bash
-bitbake-layers add-layer ~/yocto-thinedge/meta-openembedded/meta-oe
-bitbake-layers add-layer ~/yocto-thinedge/meta-openembedded/meta-python
-bitbake-layers add-layer ~/yocto-thinedge/meta-openembedded/meta-networking
-bitbake-layers add-layer ~/yocto-thinedge/meta-tedge
+bitbake-layers add-layer /home/yocto/meta-openembedded/meta-oe
```

Comment:
Please mention from where this command has to be run, I tried from the poky level, but it dint work. Tried from inside `poky/build` then it did work. Is that the right place to run this command?

Reply:
I was under the impression that it will work from any directory, as both `bitbake` and `runqemu` do, but you are correct that it has to be run from `poky/build` directory, or else the tool won't be able to locate `poky/build/conf/bblayers.conf` file. This file can also be edited manually without the tool.